### PR TITLE
[get-pixels] Create @types/get-pixels

### PR DESCRIPTION
**Pull request by donmccurdy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

***

So far I've been able to run `npm test get-pixels`, I see errors in the test scaffolding. Perhaps related to a sparse checkout, but I'm not sure.

<details>

<summary>Test output</summary>

```
❯ npm test get-pixels

> definitely-typed@0.0.3 test
> dtslint types "get-pixels"

internal/modules/cjs/loader.js:883
  throw err;
  ^

Error: Cannot find module './lib/pack.js'
Require stack:
- /Users/donmccurdy/DefinitelyTyped/node_modules/tar/tar.js
- /Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/io.js
- /Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/fs.js
- /Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/index.js
- /Users/donmccurdy/DefinitelyTyped/node_modules/dtslint/bin/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/donmccurdy/DefinitelyTyped/node_modules/tar/tar.js:171:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/donmccurdy/DefinitelyTyped/node_modules/tar/tar.js',
    '/Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/io.js',
    '/Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/fs.js',
    '/Users/donmccurdy/DefinitelyTyped/node_modules/@definitelytyped/utils/dist/index.js',
    '/Users/donmccurdy/DefinitelyTyped/node_modules/dtslint/bin/index.js'
  ]
}

```

</details>

### CODE REVIEW

File: types/get-pixels/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/get-pixels/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```

File: types/get-pixels/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for get-pixels 3.3
+// Project: https://github.com/scijs/get-pixels#readme
+// Definitions by: Don McCurdy <https://github.com/donmccurdy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import ndarray = require('ndarray');
+
+type Callback = (err: string | null, pixels: ndarray | null) => void;
```
Comment (peterblazejewicz) [inline types/get-pixels/index.d.ts:10]:
```ts
type Callback = (err: string | null, pixels: ndarray | null) => void;
```
the callback seems to support `Error`:
https://github.com/scijs/get-pixels/blob/master/node-pixels.js#L137

File: types/get-pixels/get-pixels-tests.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import getPixels from 'get-pixels';
+import ndarray from 'ndarray';
```
Comment (peterblazejewicz) [inline types/get-pixels/get-pixels-tests.ts:2]:
```suggestion
import getPixels = require('get-pixels');
import ndarray = require('ndarray');
```

File: types/get-pixels/index.d.ts (hunk 2 above)
Comment (donmccurdy) [reply types/get-pixels/index.d.ts]: Thanks! Can't see anywhere using a string after all, so I've made this `Error | null`.

**Review by peterblazejewicz — COMMENTED** [review]
cool, now remaining part. You need to review the callback, to match actual code:
https://github.com/scijs/get-pixels/blob/master/node-pixels.js#L94
https://github.com/scijs/get-pixels/blob/master/node-pixels.js#L94
it can just follow Node patter IMO:
```ts
type Callback = (err: Error | null, pixels: ndarray) => void;
```

**Comment by donmccurdy** [comment]
Hm, sorry I'm not following:

```ts
// Current:
type Callback = (err: Error | null, pixels: ndarray | null) => void;

// Proposed: 
type Callback = (err: Error | null, pixels: ndarray) => void;
```

The `pixels` argument is not guaranteed to be given, as the lines you link to show, so isn't the first type correct?

**Comment by peterblazejewicz** [comment]
> The `pixels` argument is not guaranteed to be given, as the lines you link to show, so isn't the first type correct?

I understand, but I understand we promote established patterns for handling, that one will results in TS error:
https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=18&pc=4#code/C4TwDgpgBAwghgGwQIzgYwNZQLxQBQQBOhAXFAKLED2hUAPlAHYCuSANFGAJYAeECAZzIDghLowDmAbQC69JqwQBKHAD4oANypcAJgG4AUAfHAiAM3TQYVRqPTAoAbwNQoAWwjAAFlR140yAD8ZPBIqJhKZFq6hgC+Rmg2IlBmVFQhNnZoDrjOrh7evmQBak6xbAbxBqlUAHQFPn4ExBzcfIIq2Op5UFxmzYQqPa5EhPUQAgJwEhB6UAD081AAJOQ8kNkAKuDQImKSLq5QhJ7MhIyGrvGuiytrG8CUhDRQAPLIAFYQ2b0CnFSTLjIBAgBRIQ5tfgCWoICCSbxxJR6IA
```ts
// $ExpectError Object is possibly null
  pixels.length;
```

**Comment by donmccurdy** [comment]
I see, thanks! Wasn't aware of the convention, but can understand how the ergonomics would be better this way. Pushed the change. 👍

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@donmccurdy thx!

**Review by orta — APPROVED** [review]
Looks good to me too, thanks!

**Comment by donmccurdy** [comment]
Ready to merge

**Pull request merged**